src/components/swap/ConfirmSwapModal.tsx

```diff
@@ -286,6 +288,10 @@ export default function ConfirmSwapModal({
   onCurrencySelection: (field: Field, currency: Currency) => void
   fiatValueInput: { data?: number; isLoading: boolean }
   fiatValueOutput: { data?: number; isLoading: boolean }
+  // onConfirm triggers the same callback, but it also clears other swap state.
+  // for the UniswapX signature expiration logic, we need a callback which only
+  // requests the signature without clearing swap state.
+  onRetryUniswapXSignature?: () => void
```

Comment:
possibly more usable/readable/less hacky alternative would be to provide `onconfirm` and `onclear` and call both for no uniswapx signature retry

Reply:
i like it